### PR TITLE
Add branching title

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Generate unique sequential branching names and storing into the metadata.

The PR https://github.com/ministryofjustice/fb-metadata-presenter/pull/168 needs to be merged first

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,11 +10,11 @@ gem 'delayed_job_active_record'
...
 # one of these lines:
-# gem 'metadata_presenter',
-#     github: 'ministryofjustice/fb-metadata-presenter',
-#     branch: 'page-with-component'
+gem 'metadata_presenter',
+    github: 'ministryofjustice/fb-metadata-presenter',
+    branch: 'add-branching-title'
 # gem 'metadata_presenter', path: '../fb-metadata-presenter'
-gem 'metadata_presenter', '2.2.0'
+# gem 'metadata_presenter', '2.2.0'
```
Comment (brenetic) [inline Gemfile:17]: can swap this back to `2.2.0`

File: app/services/branch_title_generator.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module BranchTitleGenerator
+  def title
+    titles = branches.map(&:title)
+    possible_title = branches.size + 1
+    name = "Branching #{possible_title}"
```
Comment (brenetic) [inline app/services/branch_title_generator.rb:5]: current design says `Branching node . . .`

File: Gemfile (hunk 1 above)
Comment (tomas-stefano) [reply Gemfile]: I was waiting for the 2.3.0 to be released. All good now 🚀

File: app/services/branch_title_generator.rb (hunk 2 above)
Comment (tomas-stefano) [reply app/services/branch_title_generator.rb]: Yes, I was waiting after standup to change and move to a locale. That's also done! 🚀 🚀 🚀

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**